Sources/Commands/SwiftTestTool.swift

```diff
@@ -288,6 +292,38 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
                 executionStatus = .failure
             }
         }
+
+        if options.shouldEnableCodeCoverage {
+            let buildParameters = try self.buildParameters()
+            let toolChain = try getToolchain()
+            try mergeRawDataFiles(fs: fileManager, path: buildParameters.buildPath.appending(component: "codecov"), toolChain: toolChain)
+            let graph = try loadPackageGraph()
+            for packages in graph.rootPackages {
+                let component = packages.name + "PackageTests"
+                let testPath = try buildTestsIfNeeded(options, graph: graph).appending(component: "Contents").appending(component: "MacOS").appending(components: component)
+                try generateReport(fs: fileManager, testPath: testPath, buildPath: buildParameters.buildPath, name: packages.name, toolChain: toolChain)
+            }
+        }
+    }
+
+    private func mergeRawDataFiles(fs: FileSystem, path: AbsolutePath, toolChain: UserToolchain) throws {
+            let files = try fs.getDirectoryContents(path)
+            let llvmProf = try toolChain.getLlvmProf()
+            var args: [String] = [llvmProf.asString, "merge", "-sparse"]
```

Comment:
I think we don't need the explicit type here.

Reply:
done